service/matching/matching_engine.go

```diff
@@ -1987,6 +2119,114 @@ func (e *matchingEngineImpl) pollTask(
 	return pm.PollTask(ctx, pollMetadata)
 }
 
+// isValidName checks if each character is a letter/number in the input string
```

Comment:
it seems cleaner to move all these stuff to a different file called `deployment_workflow_util.go` or something similar.